### PR TITLE
MacVim: Link mvim with brewed ruby & perl

### PR DESCRIPTION
**Pull request by CarterLi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Like what [vim does](https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb#L39) for long time. This requires upstream change https://github.com/macvim-dev/macvim/commit/9d23d58d4c8800b395986fbcdeba4c80e7afe4d7
Tested with `--HEAD`

Also add `gettext` option.
Note: MacVim won't bundle `.po` files by default, you may copy them into `MacVim.app/Contents/Resources/vim/runtime/lang` manually

Also fix tests when built with python3 ;)

### CODE REVIEW

**Comment by CarterLi** [comment]
Related issues: https://github.com/Homebrew/homebrew-core/issues/10824

**Comment by ilovezfs** [comment]
I have déjà vu.

File: Formula/macvim.rb [hunk 1]
```diff
@@ -14,20 +14,29 @@ class Macvim < Formula
   end
 
   option "with-override-system-vim", "Override system vim"
+  option "with-gettext", "Build MacVim with +gettext."
 
   deprecated_option "override-system-vim" => "with-override-system-vim"
 
+  conflicts_with "vim",
+    :because => "They both override system vim" if build.with? "override-system-vim"
```
Comment (ilovezfs) [inline Formula/macvim.rb]: This won't work properly unless you also inspect the `tab` (INSTALL_RECEIPT) of vim to see whether it was built with `override-system-vi`.

Comment (CarterLi) [inline Formula/macvim.rb]: Huh?
[override-system-vi](https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb#L115) only generates `symlink "vim" => "vi"`, system global symlink `vim` is always generated whatever override-system-vi or not.

Comment (ilovezfs) [inline Formula/macvim.rb]: Yes you're right.

**Comment by ilovezfs** [comment]
I suspect these changes are likely to be quite controversial since you're adding rather heavy new dependencies here.

**Comment by CarterLi** [comment]
I can make brewed ruby & perl optional @ilovezfs

**Comment by ilovezfs** [comment]
Let's see what other people think.

File: Formula/macvim.rb [hunk 2]
```diff
@@ -14,20 +14,29 @@ class Macvim < Formula
   end
 
   option "with-override-system-vim", "Override system vim"
+  option "with-gettext", "Build MacVim with +gettext."
```
Comment (dunn) [inline Formula/macvim.rb]: The `depends_on "gettext" => :optional` below will generate the option text, so we don't need this line.

File: Formula/macvim.rb [hunk 3]
```diff
@@ -14,20 +14,29 @@ class Macvim < Formula
...
 
   deprecated_option "override-system-vim" => "with-override-system-vim"
 
+  conflicts_with "vim",
```
Comment (dunn) [inline Formula/macvim.rb]: It's been a while, but [last I checked](https://github.com/Homebrew/legacy-homebrew/pull/42314) we couldn't do conditional formulae conflicts.  Has that changed?

Comment (ilovezfs) [inline Formula/macvim.rb]: @dunn I don't know if anything has changed, but look at the current version `moreutils` in core. It's baaaack.

File: Formula/macvim.rb (hunk 2 above)
Comment (CarterLi) [inline Formula/macvim.rb]: Removed

File: Formula/macvim.rb (hunk 3 above)
Comment (dunn) [inline Formula/macvim.rb]: D:

**Comment by chdiza** [comment]
> I suspect these changes are likely to be quite controversial since you're adding rather heavy new dependencies here.

This.  Don't add those, please.

**Comment by CarterLi** [comment]
Upstream have released [a new version](https://github.com/macvim-dev/macvim/releases/tag/snapshot-128);this PR is ready for merging.
For new dependencies: [vim formula can](https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb#L39), why MacVim can't?

**Comment by ilovezfs** [comment]
@DomT4 any thoughts here?

**Comment by DomT4** [comment]
On the one hand I lean towards @chdiza's & your position pretty strongly, but on the other hand when this was done to `vim` I think we got <15 complaints out of the thousands of people who install `vim` regularly, so. I think _most people_ care less here than we expected, and I think you'd get similar results again.

The backticks in `test do` are nasty though 😝.

**Comment by ilovezfs** [comment]
Fits of domitus. This is definitely encouraging.

**Comment by chdiza** [comment]
> For new dependencies: vim can, why MacVim can't?

Vim should not have ever been made to depend on ruby (or any other scripting language, for that matter).

Also, MacVim is historically much more finicky than Vim about building against Ruby, both in and out of Homebrew.

**Comment by CarterLi** [comment]
I've made dependencies `ruby` & `perl` optional.
Any chance to merge?

**Comment by thefonso** [comment]
@CarterLi  this crazy sauce with vim and macvim has persisted for years. The macvim team seems to be stuck. If you cats on homebrew have a workable solution, I hope you can make this happen! ^_^

**Pull request closed without merging**